html/dom/dynamic-markup-insertion/document-write/contentType.window.js

```diff
@@ -0,0 +1,27 @@
+// META: script=/common/media.js
+
+const videoURL = getVideoURI("/images/pattern"),
+      videoMIMEType = getMediaContentType(videoURL);
+
+[
+  [videoURL, videoMIMEType],
+  ["/images/red.png", "image/png"],
+  ["/common/text-plain.txt", "text/plain"],
+  ["/common/blank.html", "text/html"]
+].forEach(val => {
+  async_test(t => {
+    const frame = document.body.appendChild(document.createElement("iframe"));
+    t.add_cleanup(() => frame.remove());
+    frame.src = val[0];
+    frame.onload = t.step_func_done(() => {
+      assert_equals(frame.contentDocument.contentType, val[1]);
+      frame.contentDocument.write("<b>Heya</b>");
+      assert_equals(frame.contentDocument.body.firstChild.localName, "b");
+      assert_equals(frame.contentDocument.body.firstChild.textContent, "Heya");
+      assert_equals(frame.contentDocument.contentType, val[1]);
+
+      // Make sure a load event is fired across browsers
+      frame.contentDocument.close();
+    });
+  }, "document.write(): " + val[1] + " document");
```

Comment:
For the video, this will result in a different test name on different browsers. It'd be extra work to not have that be the case, so leaving it alone until we figure out a "rule" and how to enforce it seems right.